tgui/packages/tgui/interfaces/Cargo/CargoCatalog.tsx

```diff
@@ -101,143 +93,254 @@ type CatalogTabsProps = {
-                  // return to normal category
-                  setActiveSupplyName(sorted[0]?.name);
-                }
-                setSearchText(value);
-              }}
-            />
-          </Stack.Item>
-        </Stack>
-      </Tabs.Tab>
-
-      {sorted.map((supply) => (
-        <Tabs.Tab
-          className="candystripe"
-          color={supply.name === activeSupplyName ? 'green' : undefined}
-          key={supply.name}
-          selected={supply.name === activeSupplyName}
-          onClick={() => {
-            setActiveSupplyName(supply.name);
-            setSearchText('');
+    <Stack fill vertical>
+      <Stack.Item>
+        <SearchBar
+          query={searchText}
+          onSearch={(value) => {
+            if (value === searchText) {
+              return;
+            }
+
+            if (value.length) {
+              // Start showing results
+              setActiveSupplyName('search_results');
+            } else if (activeSupplyName === 'search_results') {
+              // return to normal category
+              setActiveSupplyName(sorted[0]?.name);
+            }
+            setSearchText(value);
           }}
-        >
-          <div style={{ display: 'flex', justifyContent: 'space-between' }}>
-            <span>{supply.name}</span>
-            <span> {supply.packs.length}</span>
-          </div>
-        </Tabs.Tab>
-      ))}
-    </Tabs>
+        />
+      </Stack.Item>
+      <Stack.Item grow p={1} m={-1} mt={1} overflowY={'auto'}>
+        <Tabs vertical>
+          <Tabs.Tab
+            key="search_results"
+            selected={activeSupplyName === 'search_results'}
+            style={{ display: 'none' }}
+          />
+
+          {sorted.map((supply) => (
+            <Tabs.Tab
+              className="candystripe"
+              color={supply.name === activeSupplyName ? 'green' : undefined}
+              key={supply.name}
+              selected={supply.name === activeSupplyName}
+              onClick={() => {
+                setActiveSupplyName(supply.name);
+                setSearchText('');
+              }}
+            >
+              <Stack justify={'space-between'}>
+                <span>{supply.name}</span>
+                <span> {supply.packs.length}</span>
+              </Stack>
+            </Tabs.Tab>
+          ))}
+        </Tabs>
+      </Stack.Item>
+      <Stack.Item>
+        {!express && (
+          <Button
+            fluid
+            color={self_paid ? 'caution' : 'transparent'}
+            icon={self_paid ? 'check-square-o' : 'square-o'}
+            onClick={() => act('toggleprivate')}
+            tooltip={'Use your own funds to purchase items.'}
+            tooltipPosition={'top'}
+          >
+            Buy Privately
+          </Button>
+        )}
+      </Stack.Item>
+    </Stack>
   );
 }
 
 type CatalogListProps = {
   packs: SupplyCategory['packs'];
+  openContents: Dispatch<string>;
```

Comment:
Is this a state setter? I believe the type is wrong, it would be Dispatch<SetStateAction<string>>